lms/djangoapps/courseware/grades.py

```diff
@@ -25,10 +26,71 @@
 from .module_render import get_module_for_descriptor
 from submissions import api as sub_api  # installed from the edx-submissions repository
 from opaque_keys import InvalidKeyError
+from xblock.core import XBlock
 
 
 log = logging.getLogger("edx.courseware")
 
+GRADED_LOCATION_CATEGORIES = set(
+    cat for (cat, xblock_class) in XBlock.load_classes() if (
+        getattr(xblock_class, 'has_score', True) or getattr(xblock_class, 'has_children', True)
+    )
+)
+
+
+def descriptor_filter(descriptor):
+    """
+    Pass a location filter to the field data cache so that we don't
+    instantiate descriptors that are never graded (like video descriptors)
+    during grading
+    """
+    return descriptor.location.category in GRADED_LOCATION_CATEGORIES
+
+
+class MaxScoresCache(object):
+    """
+    A cache for students' problems' max scores when students look at, but
+    do not answer, problems.
+    """
+    def __init__(self, locations):
+        self.locations = locations
+        self._max_scores_cache = {}
+        self._max_scores_updates = {}
+
+    def fetch_from_remote(self):
+        """
+        Populate the local cache with values from django's cache
+        """
+        self._max_scores_cache = cache.get_many(
+            [unicode(loc) for loc in self.locations]
```

Comment:
Minor: Since it seems like the only place `self.locations` is used is here, does it make sense to just make `fetch_from_remote()` take a list of locations as an arg?